Description: Since */src/freemarker/ext/jsp/*.dtd and
 */src/freemarker/ext/jsp/*.xsd has been removed from orig tarball,
 resolve those files using classpath (into jsp-api-2.1.jar
 and servlet-api-2.5.jar)
Author: Damien Raude-Morvan <drazzib@debian.org>
Last-Update: 2011-09-22
Forwarded: no
--- a/src/main/java/freemarker/ext/jsp/TaglibFactory.java
+++ b/src/main/java/freemarker/ext/jsp/TaglibFactory.java
@@ -1870,17 +1870,17 @@
         static
         {
             // JSP taglib 1.2
-            DTDS.put("-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.2//EN", "web-jsptaglibrary_1_2.dtd");
-            DTDS.put("http://java.sun.com/dtd/web-jsptaglibrary_1_2.dtd", "web-jsptaglibrary_1_2.dtd");
+            DTDS.put("-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.2//EN", "/javax/servlet/jsp/resources/web-jsptaglibrary_1_2.dtd");
+            DTDS.put("http://java.sun.com/dtd/web-jsptaglibrary_1_2.dtd", "/javax/servlet/jsp/resources/web-jsptaglibrary_1_2.dtd");
             // JSP taglib 1.1
-            DTDS.put("-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.1//EN", "web-jsptaglibrary_1_1.dtd");
-            DTDS.put("http://java.sun.com/j2ee/dtds/web-jsptaglibrary_1_1.dtd", "web-jsptaglibrary_1_1.dtd");
+            DTDS.put("-//Sun Microsystems, Inc.//DTD JSP Tag Library 1.1//EN", "/javax/servlet/jsp/resources/web-jsptaglibrary_1_1.dtd");
+            DTDS.put("http://java.sun.com/j2ee/dtds/web-jsptaglibrary_1_1.dtd", "/javax/servlet/jsp/resources/web-jsptaglibrary_1_1.dtd");
             // Servlet 2.3
-            DTDS.put("-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN", "web-app_2_3.dtd");
-            DTDS.put("http://java.sun.com/dtd/web-app_2_3.dtd", "web-app_2_3.dtd");
+            DTDS.put("-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN", "/javax/servlet/resources/web-app_2_3.dtd");
+            DTDS.put("http://java.sun.com/dtd/web-app_2_3.dtd", "/javax/servlet/resources/web-app_2_3.dtd");
             // Servlet 2.2
-            DTDS.put("-//Sun Microsystems, Inc.//DTD Web Application 2.2//EN", "web-app_2_2.dtd");
-            DTDS.put("http://java.sun.com/j2ee/dtds/web-app_2_2.dtd", "web-app_2_2.dtd");
+            DTDS.put("-//Sun Microsystems, Inc.//DTD Web Application 2.2//EN", "/javax/servlet/resources/web-app_2_2.dtd");
+            DTDS.put("http://java.sun.com/j2ee/dtds/web-app_2_2.dtd", "/javax/servlet/resources/web-app_2_2.dtd");
         }
         
         public InputSource resolveEntity(String publicId, String systemId)
